### PR TITLE
Feature/oauth auth z

### PR DESCRIPTION
**Pull request by bsekar**

### CODE REVIEW

File: src/main/java/jetbrains/buildServer/auth/oauth/ServerPrincipalFactory.java [hunk 1]
```diff
@@ -32,53 +34,94 @@ public ServerPrincipalFactory(@NotNull UserModel userModel, @NotNull UserGroupMa
...
             final SUser user = userModel.findUserByUsername(userName, PluginConstants.ID_USER_PROPERTY_KEY);
-            return Optional.ofNullable(user).map(u -> new ServerPrincipal(PluginConstants.OAUTH_AUTH_SCHEME_NAME, user.getUsername()));
+            return Optional.ofNullable(user);
         } catch (InvalidUsernameException e) {
             // ignore it
             return Optional.empty();
         }
     }
+
+    private List<String> obtainGroupsDelta(List<String> minuend, List<String> subtrahend) {
```
Comment (bsekar) [inline src/main/java/jetbrains/buildServer/auth/oauth/ServerPrincipalFactory.java:117]: This method identifies the group names in the first arg, that are not in the second. Can't think of a better name for it. @cwrm56

Comment (cwrm56) [reply src/main/java/jetbrains/buildServer/auth/oauth/ServerPrincipalFactory.java:117]: The method name is fine, but I had to lookup what minuend and subtrahend mean!

File: src/main/resources/buildServerResources/editOAuthSchemeProperties.jsp [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <%@ page import="jetbrains.buildServer.auth.oauth.ConfigKey" %>
 <%@ include file="/include-internal.jsp" %>
 <%@ taglib prefix="prop" tagdir="/WEB-INF/tags/props" %>
-<%@ taglib prefix="c" uri="http://java.sun.com/jsp/jstl/core" %>
+<%--<%@ taglib prefix="c" uri="http://java.sun.com/jsp/jstl/core" %>--%>
```
Comment (cwrm56) [inline src/main/resources/buildServerResources/editOAuthSchemeProperties.jsp]: Mark added this line, fine to remove it

File: src/main/resources/buildServerResources/editOAuthSchemeProperties.jsp [hunk 3]
```diff
@@ -21,13 +21,25 @@
...
+                    groupSettings.show();
                 } else {
                     settings.hide();
+                    groupSettings.hide();
                 }
             },
+            showOrHideGroups: function () {
+              var groupsListDiv = $j('#groups_list');
+              var syncGroups = $j('#<%=ConfigKey.syncGroups.toString()%>');
+              if(syncGroups[0].checked) {
```
Comment (cwrm56) [inline src/main/resources/buildServerResources/editOAuthSchemeProperties.jsp]: Formatting - space after if.

File: src/main/java/jetbrains/buildServer/auth/oauth/AuthenticationSchemeProperties.java [hunk 4]
```diff
@@ -62,15 +62,24 @@ public String getClientSecret() {
     public String getScope() {
         return getProperty(ConfigKey.scope);
     }
+//
```
Comment (cwrm56) [inline src/main/java/jetbrains/buildServer/auth/oauth/AuthenticationSchemeProperties.java]: Remove this

**Pull request merged**